core/shared/platform/windows/platform_internal.h

```diff
@@ -137,9 +140,47 @@ bh_atomic_thread_fence(int mem_order);
 
 #define os_atomic_thread_fence bh_atomic_thread_fence
 
-typedef HANDLE os_handle;
-typedef void os_dir_stream;
+typedef enum windows_handle_type {
+    windows_handle_type_socket,
+    windows_handle_type_normal,
+    windows_handle_type_stdio
+} windows_handle_type;
+
+typedef enum windows_access_mode {
```

Comment:
more often we use flags for this, e.g.:
typedef enum {
  read = 1 << 1,
  write = 1 << 2,
} mode;
```
so you not only have all the combinations, but also can easily perform checks, e.g.
```
// can I write?
if (socket->mode & write > 0) {
  // yes I can
}
```
instead of
```
if (socket->mode == read_only || socket->mode == read_write) {
 //...
}
```

Reply:
thanks, that makes sense! Bitflags would be better indeed.